src/instamatic/experiments/fast_adt/experiment.py

```diff
@@ -0,0 +1,452 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass, field
+from pathlib import Path
+from queue import Queue
+from threading import Thread
+from typing import Any, Dict, Iterator, List, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import pandas as pd
+from matplotlib import pyplot as plt
+from typing_extensions import Self
+
+from instamatic import config
+from instamatic._collections import NullLogger
+from instamatic._typing import AnyPath
+from instamatic.calibrate import CalibBeamShift, CalibMovieDelays, CalibStageRotation
+from instamatic.calibrate.filenames import CALIB_BEAMSHIFT
+from instamatic.experiments.experiment_base import ExperimentBase
+from instamatic.processing.ImgConversionTPX import ImgConversionTPX as ImgConversion
+
+
+def safe_range(start: float, stop: float, step: float) -> np.ndarray:
```

Comment:
For safety, I normally force kwargs only in functions:

```suggestion
def safe_range(*, start: float, stop: float, step: float) -> np.ndarray:
```